File: LABA5/commandsPack/Command.java
package commandsPack;


import mainPack.MovieCollection;

import java.util.Scanner;

public abstract class Command {
    protected Scanner reader;
    protected MovieCollection collection;
    protected boolean readingFromFile = false;

    public void execute() {

    }

    public Scanner getReader() {
        return reader;
    }

    public void setReader(Scanner reader) {
        this.reader = reader;
    }

    public void setCollection(MovieCollection collection) {
        this.collection = collection;
    }
}

File: LABA5kt/src/commandsPack/Command.kt
package commandsPack


//import com.thoughtworks.xstream.XStream
import mainPack.MovieCollection
import java.util.*

/**
 * Abstract class inherited by all the commands available
 */
abstract class Command protected constructor() {
    protected var reader: Scanner? = null
    protected var collection: MovieCollection? = null
    //protected var xStream: XStream? = null
    protected var readingFromFile: Boolean = false
    protected var fileCallsCount: Int = 0

    val nextReaderLine: String
        /**
         * @return returns next line of scanner
         */
        get() = reader!!.nextLine()
    val nextReaderWord: String
        get() = reader!!.next()

    fun setReader(reader: Scanner?) {
        this.reader = reader
    }

    fun setCollection(collection: MovieCollection?) {
        this.collection = collection
    }
}